### PR TITLE
Add support for table_delta_v1

### PR DESCRIPTION
**Pull request by jgiszczak**
Add support for table_delta_v1.

### CODE REVIEW

File: src/fill_pg_plugin.cpp [hunk 1]
```diff
@@ -711,33 +711,37 @@ struct fpg_session : connection_callbacks, std::enable_shared_from_this<fpg_sess
...
-            }
-            numRows += table_delta.rows.size();
+            auto& variant_type = get_type(std::visit([](auto&& arg){return arg.name;}, t_delta));
+            if (!variant_type.as_variant() || variant_type.as_variant()->size() != 2 || !variant_type.as_variant()->at(0).type->as_struct() || !variant_type.as_variant()->at(1).type->as_struct())
+                throw std::runtime_error("don't know how to process " + variant_type.name);
+
+            std::visit([&block_num, &bulk, &t, &pipeline, this](auto t_delta){
+               size_t num_processed = 0;
+               auto& variant_type = get_type(t_delta.name);
+               auto& type = *variant_type.as_variant()->at(0).type;
```
Comment (jgiszczak) [inline src/fill_pg_plugin.cpp:729]: I am unsure if this is correct.

File: src/fill_pg_plugin.cpp [hunk 2]
```diff
@@ -711,33 +711,37 @@ struct fpg_session : connection_callbacks, std::enable_shared_from_this<fpg_sess
...
-                std::string fields = "block_num, present";
-                std::string values = std::to_string(block_num) + sep(bulk) + sql_str(bulk, row.present);
-                for (auto& field : type.as_struct()->fields)
-                    fill_value(bulk, false, t, "", fields, values, row.data, field);
-                write(block_num, t, pipeline, bulk, table_delta.name, fields, values);
-                ++num_processed;
-            }
-            numRows += table_delta.rows.size();
+            auto& variant_type = get_type(std::visit([](auto&& arg){return arg.name;}, t_delta));
+            if (!variant_type.as_variant() || variant_type.as_variant()->size() != 2 || !variant_type.as_variant()->at(0).type->as_struct() || !variant_type.as_variant()->at(1).type->as_struct())
```
Comment (jgiszczak) [inline src/fill_pg_plugin.cpp:723]: Should these variant checks stay?

File: src/fill_pg_plugin.cpp [hunk 3]
```diff
@@ -711,33 +711,37 @@ struct fpg_session : connection_callbacks, std::enable_shared_from_this<fpg_sess
         unsigned numRows = 0;
         varuint32_from_bin(num, bin);
         for (uint32_t i = 0; i < num; ++i) {
-            check_variant(bin, get_type("table_delta"), "table_delta_v0");
-            table_delta_v0 table_delta;
-            from_bin(table_delta, bin);
+            uint32_t index;
+            eosio::varuint32_from_bin(index, bin);
+            table_delta t_delta;
+            from_bin(t_delta, bin);
```
Comment (tbfleming) [inline src/fill_pg_plugin.cpp:717]: This double-reads the variant's index, causing corruption. `index` isn't needed.

File: src/fill_pg_plugin.cpp (hunk 2 above)
Comment (tbfleming) [reply src/fill_pg_plugin.cpp:723]: The`variant_type.as_variant()->size()` checks the inner data's variant (account/account_metadata/code/contract_table/etc), not the `table_delta` variant. Of these, only `chain_config` and `global_property` have size 2. `global_property` was already skipped. You'll need to either:

* skip `chain_config`
* add special handling for `chain_config` and maybe `global_property`
* come up with a general-purpose way of mapping variants to pg's inflexible column structure

File: src/fill_pg_plugin.cpp (hunk 3 above)
Comment (jgiszczak) [reply src/fill_pg_plugin.cpp:717]: The varuint32_from_bin is replacing a read in this location formerly done by `check_variant`.  Was that an unexpected read?

File: src/fill_pg_plugin.cpp (hunk 1 above)
Comment (cj-oci) [reply src/fill_pg_plugin.cpp:729]: Looks right.

File: src/fill_pg_plugin.cpp (hunk 3 above)
Comment (tbfleming) [reply src/fill_pg_plugin.cpp:717]: The previous code:
* `check_variant` (read the index)
* read a `table_delta_v0` (not a variant - doesn't read the index)

Comment (jgiszczak) [reply src/fill_pg_plugin.cpp:715]: Understood.  Removed.

File: src/fill_pg_plugin.cpp (hunk 2 above)
Comment (jgiszczak) [reply src/fill_pg_plugin.cpp]: Elected to skip `chain_config`.

Created https://blockone.atlassian.net/browse/EPE-517 to document need to support `chain_config` and `global_property`.

**Review by tbfleming — APPROVED** [review]

**Pull request merged**